src/main/java/cafe/cryptography/curve25519/EdwardsPoint.java

```diff
@@ -263,7 +263,38 @@ public EdwardsPoint multiply(final Scalar s) {
      */
     public static EdwardsPoint vartimeDoubleScalarMultiplyBasepoint(final Scalar a, final EdwardsPoint A,
             final Scalar b) {
-        throw new UnsupportedOperationException();
+        final byte[] aNaf = a.nonAdjacentForm();
+        final byte[] bNaf = b.nonAdjacentForm();
+
+        ProjectiveNielsPoint.NafLookupTable tableA = ProjectiveNielsPoint.buildNafLookupTable(A);
+        AffineNielsPoint.NafLookupTable tableB = AffineNielsPoint.buildNafLookupTable(Constants.ED25519_BASEPOINT);
```

Comment:
`AffineNielsPoint` is used for the Ed25519 basepoint because it is slower to build the table (requiring inversions) but faster to do point addition and subtraction with once computed (one fewer FieldElement multiplications). This will be amortized out later by storing this lookup table as a constant (either computing it when the library is loaded, or ideally hard-coding the table itself).